doc: update darwin uninstall instructions (#6678)
NixOS/nix: doc/manual/src/installation/installing-binary.md

   This will remove all the build users that no longer serve a purpose.

4. Edit fstab using `sudo vifs` to remove the line mounting the Nix Store
   volume on `/nix`, which looks like

>    `UUID=<unique uuid> /nix apfs rw,noauto,nobrowse,suid,owners` or

Reviewer: ```suggestion
   `UUID=<uuid> /nix apfs rw,noauto,nobrowse,suid,owners` or
```

UUIDs are already unique.

   This will remove the Nix Store volume and everything that was added to the
   store.

   If this command indicates that it couldn't remove the volume, your Nix Store
   volume may not be mounted. Run the following to double-check:

   ```console
   diskutil list
   ```

>    If you see a Nix Store volume, re-run the diskutil deleteVolume command, but

Reviewer 2: This explains what to do if the volume is indeed mounted but not what to do if it's not. Probably as simple as mounting it but maybe there's some help we can offer here on how to figure out how to find the right volume to mount?

Reviewer 2: Ah, I was the one who misunderstood. The "to double-check" made it sound to me like "it's probably mounted but the label may differ, use diskutil list to figure out the volume and used the bsd identifier."

Maybe it can say something along the lines of, "If there's no Nix Store volume, you can skip this step. Otherwise use the identifier to delete it?"

Author: I've force-pushed some phrasing tweaks. WDYT?

Reviewer 2: Perfect!

   This will remove the Nix Store volume and everything that was added to the
   store.

>    If this command indicates that it couldn't remove the volume, you should

Reviewer: ```suggestion
   If the output indicates that the command couldn't remove the volume, you should
```

<!-- Note: anchors above to catch permalinks to old explanations -->

We believe we have ironed out how to cleanly support the read-only root

> on modern macOS. New installs will do this automatically.

Author: To clarify, the old statement was true for a while (and is technically true of the volume-setup parts of the macOS install), but installs can now break with a populated store since `rsync` was swapped out for `cp` in #5150. (I opened https://github.com/NixOS/nix/issues/6679 so that we have a chance to avoid losing the knowledge that it once worked this way.)

We've been getting a steady trickle of these reports since ~December (https://github.com/NixOS/nix/search?o=desc&q=%22cp%3A+cannot+overwrite+directory%22&s=created&type=issues). Not that I think fixing the language will actually help (I imagine most of these people are just _reasoning_ that a reinstall can/would/should fix whatever ails them).
